### PR TITLE
Mobile designs

### PR DESCRIPTION
**Pull request by justagoodfriend**
Updates Attendance Change Modal to better match Serenas Figma
Updates Attendance Change Modal to be shown on Desktop
Adds in Additional Cases of Attendance Status on the Attendance Record Page

Images: 
Screen Shot 2023-12-03 at 10 41 00 PM
Screen Shot 2023-12-03 at 10 38 30 PM
Screen Shot 2023-12-03 at 10 38 50 PM
Screen Shot 2023-12-03 at 10 39 54 PM
Screen Shot 2023-12-03 at 10 40 21 PM

### CODE REVIEW

File: src/components/AttendanceRecord/AttendanceStanding.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  const amtEvents = attendanceRecord.length;
+  const unexcusedA = getCountOfKeyInStatusList(AttendanceList, "A");
+  const unexcusedK = getCountOfKeyInStatusList(AttendanceList, "K");
+  const absent = unexcusedA + unexcusedK;
+  const tooManyAbsent = absent >= 2;
+  const excusedAbsences = excusedAbsenceCount(AttendanceList);
+  const tooManyExcused =
+    ((absent + excusedAbsences) / amtEvents) * 100 > 50 && amtEvents > 4;
+
+  const createStrings = () => {
```
Comment (justagoodfriend) [inline src/components/AttendanceRecord/AttendanceStanding.tsx]: What is a good way to do something like this, where I have four different possibilities for text/className for the same button, this seems pretty bad style although it works

Comment (naajaw) [reply src/components/AttendanceRecord/AttendanceStanding.tsx]: I'd make and enum, and then a record to map that enum to the relevant config object. Arrays are fine, the problem is 1) lack of names for indices; less readable, and 2) if the order of the config array:(`AttendanceStandingOptions`) changes, this breaks.

so, make it an enum + record: 
```typescript
export enum AttendanceStandingOption {
    REMOVED: "REMOVED",
    ON_HOLD: "ON_HOLD",
    AT_RISK: "AT_RISK",
    GOOD_STANDING: "GOOD_STANDING",
}

export const AttendanceStandingOptions: Record<AttendanceStandingOption, AttendanceStandingDisplay> = [
  [AttendanceStandingOption.REMOVED]: {
    text: "Removed",
    className:
      "bg-attendance-grey border border-attendance-grey bg-opacity-25 text-atn-disabled-text",
  },
  [AttendanceStandingOption.ON_HOLD]: {
    text: "On hold",
    className:
      "bg-attendance-yellow border border-attendance-yellow bg-opacity-25 text-attendance-text-yellow",
  },
  [AttendanceStandingOption.AT_RISK]: {
    text: "At risk",
    className:
      "bg-attendance-red border border-attendance-red bg-opacity-25 text-red-500",
  },
  [AttendanceStandingOption.GOOD_STANDING]: {
    text: "Good standing",
    className:
      "bg-attendance-green border border-attendance-green bg-opacity-25 text-attendance-text-green",
  },
];
```

Then, back in this file:
```typescript
const getStanding = () => {
   if (!poorStanding) {
      return AttendanceStandingOption.REMOVED;
    } else if (pendingDiscussion) {
      return AttendanceStandingOption.ON_HOLD;
    } else if (tooManyExcused || tooManyAbsent) {
      return AttendanceStandingOption.AT_RISK;
    } else {
      return AttendanceStandingOption.GOOD_STANDING;
    }
}

const display = AttendanceStandingOptions[getStanding()];

```

Comment (naajaw) [reply src/components/AttendanceRecord/AttendanceStanding.tsx]: By doing this, you're separating the business logic of determining which class of Attendance Standing is applicable from the presentational logic of mapping that Attendance Standing to a certain classname or text content

Comment (justagoodfriend) [reply src/components/AttendanceRecord/AttendanceStanding.tsx]: Brilliant

**Review by RaveeshM — APPROVED** [review]
absolutely beautiful code  -- lgtm

**Pull request merged**